lib/trento/sap_systems/sap_system.ex

```diff
@@ -88,11 +88,18 @@ defmodule Trento.SapSystems.SapSystem do
     field :deregistered_at, :utc_datetime_usec, default: nil
 
     embeds_many :instances, Instance
+
+    # The current steam is composed by legacy events, coming from database events
```

Comment:
This is the part that I dislike a bit, but we are forced to have such a thing.
The deregistration process manager doesn't have upcast/upgrade option, so we cannot change its inner state.
This means, that when old events populate the `application_instances` list with IDs that now belong to databases, when the deregistration request arrives, the `DeregisterApplicationInstance` is sent with an ID that now belongs to a database.
Anyway, `commanded` doesn't differenciate the events per type, so it tries to apply the events for that database to the `SapSystem` aggregate, so we need to handle them. Otherwise, we have the classic pattern matching error in the `apply` function.
This `legacy` fields is used to see if the aggregate receives some database related event, and if this is the case, as the aggregate is "invalid" we basically discard all the incoming commands now on.

This only happens when we have old events

Reply:
At the end I have changed this, so the aggregate stays intact.
The issue is now handled in the enrichment protocol, which takes care of not sending deregistration commands if the ID belongs now to a database.
It is cleaner, and it looks better.
And if in the future, we clean up the event store in some migration, it is as simple as removing the module itself